cdap-proto/src/main/java/co/cask/cdap/proto/id/ArtifactId.java

```diff
@@ -25,14 +25,14 @@
 /**
  * Uniquely identifies an artifact.
  */
-// TODO: handle duplication with ArtifactId
-public class NamespacedArtifactId extends ElementId implements NamespacedId, ParentedId<NamespaceId> {
+// TODO: handle duplication with ArtifactId in cdap-api
+public class ArtifactId extends EntityId implements NamespacedId, ParentedId<NamespaceId> {
```

Comment:
Also, `ArtifactId` needs to be in cdap-api it looks like. I'll need to do some rework on `ArtifactId`.